src/blockstore/log_test.go

```diff
@@ -0,0 +1,183 @@
+//
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package blockstore
+
+import (
+	"log"
+	"testing"
+)
+
+func TestEncryptDecrypt(t *testing.T) {
+	var b1, b2, b3 *Block
+
+	log.Println("Log -- TestEncryptDecrypt -- Testing with a random block.")
+
+	b1 = GetRandomBlock()
+	b2 = GetZeroBlock()
+	b3 = GetZeroBlock()
+
+	if b1 == nil || b2 == nil || b3 == nil {
+		t.Fatal("Could not allocate block")
+	}
+
+	encryptBlock(b2, b1)
+	// Testing the dummy encryption, which is identical to source.
+	for i, v := range b1 {
+		if b2[i] != v {
+			t.Error("Encrypted block does not match to original.")
+			break
+		}
+	}
+
+	decryptBlock(b3, b2)
+	for i, v := range b1 {
+		if b3[i] != v {
+			t.Error("Decrypted block does not match to original.")
+			break
+		}
+	}
+
+}
+
+func TestHash(t *testing.T) {
+	s := []byte("TEST")
+	ref := []byte{152, 72, 22, 253, 50, 150, 34, 135, 110, 20, 144, 118, 52, 38, 78, 111, 51, 46, 159, 179}
```

Comment:
why these particular values?

Reply:
I added a concrete sample, which hash is not calculated in the tester function the same way it is calculated in the program. Not sure how else a hash wrapper function can be tested.

Reply:
Makes sense